src/lib.rs

```diff
@@ -0,0 +1,160 @@
+use lazy_static::lazy_static;
+
+use guest::prelude::*;
+use kubewarden_policy_sdk::wapc_guest as guest;
+
+use k8s_openapi::api::core::v1 as apicore;
+
+extern crate kubewarden_policy_sdk as kubewarden;
+use kubewarden::{logging, protocol_version_guest, request::ValidationRequest, validate_settings};
+
+mod settings;
+use settings::Settings;
+
+use slog::{o, warn, Logger};
+
+lazy_static! {
+    static ref LOG_DRAIN: Logger = Logger::root(
+        logging::KubewardenDrain::new(),
+        o!("policy" => "sample-policy")
+    );
+}
+
+#[no_mangle]
+pub extern "C" fn wapc_init() {
+    register_function("validate", validate);
+    register_function("validate_settings", validate_settings::<Settings>);
+    register_function("protocol_version", protocol_version_guest);
+}
+
+fn mutate_request(pvc: apicore::PersistentVolumeClaim, settings: &Settings) -> CallResult {
+    warn!(
+        LOG_DRAIN,
+        "storage class \"{}\" is not allowed, using fallback storage class",
+        settings.fallback_storage_class.clone().unwrap_or_default()
+    );
```

Comment:
All the debugging messages involve host callbacks, which can slow things down a little. I think we can definitely remove all the logging from this policy since it's not that valuable. That includes removing also the `slog` and `lazy_static` dependencies